planemo/galaxy/config.py

```diff
@@ -1312,7 +1312,7 @@ def _handle_job_config_file(config_directory, server_name, kwds):
 
 
 def _write_tool_conf(ctx, tool_paths, tool_conf_path):
-    tool_definition = _tool_conf_entry_for(tool_paths)
+    tool_definition = _tool_conf_entry_for(set(tool_paths))
```

Comment:
Why do you need this?

Reply:
Not needed, but we're ending up with a whole lot of duplicates in the tool_conf.xml file. Good question though on where they're coming from ...

Reply:
I think they are coming from [_all_tool_paths()](https://github.com/galaxyproject/planemo/blob/master/planemo/galaxy/config.py#L535). Probably a better fix would be to change that function to return a set of paths instead of a list?